### DISCUSSION TITLE
Return only filtered records for nested filter

### DISCUSSION THREAD

**Discussion by mubaidr** [prisma/prisma#13615] [Q&A] [answered]
Example: For following query

Get all User records where all posts are published and at least one related Post mentions Prisma

``` ts
const result = await prisma.user.findMany({
  where: {
    post: {
      every: {
        published: true
      },
      some: {
        content: {
          contains: "Prisma"
        }
      }
    }
  }
}
```

But this is includes all `posts` for `user`.

What is the proper way to filter nested relation and get only filtered childs in result.

**Comment by austincrim** [comment]
Hey @mubaidr,

Could you share your full schema?

**Reply by mubaidr** [reply]
> Hey @mubaidr,
> 
> Could you share your full schema?

I am using Prisma docs example scheme. User, Post etc

More explanation: 
I want to filter user by posts(published only) and results should contain users and their publishrd posts only.

**Comment by austincrim** [comment, marked as answer]
Would something like this work?
```ts
let result = await prisma.user.findMany({
  where: {
    posts: {
      every: {
        published: true
      }
    }
  },
  include: {
    posts: {
      where: {
        content: {
          contains: 'Prisma'
        }
      }
    }
  }
})
```

This should fetch the users who have all posts published and only fetch the posts that contain `Prisma`. The important concept here is that the top-level `where` will filter the retrieved `Users` while the `where` nested in the `include` will filter the `Posts`.

**Comment by mubaidr** [comment]
Did not know we can filter using `include`. Thanks

**Comment by mubaidr** [comment]
![Untitled](https://user-images.githubusercontent.com/2222702/172099944-b165abea-b6d0-440f-9a83-32fdc45070cc.png)

Can you guide me why `where` may not be available in my case? I only see `select` and `include` as shown in above screenshot.

**Reply by mubaidr** [reply]
> ![Untitled](https://user-images.githubusercontent.com/2222702/172099944-b165abea-b6d0-440f-9a83-32fdc45070cc.png)
> 
> Can you guide me why `where` may not be available in my case? I only see `select` and `include` as shown in above screenshot.

Nevermind. In this case `application` and `login` has one to one relationship, that is why `where` filter is not available.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
